content/post/2023-08-09-data-transfer.md

```diff
@@ -0,0 +1,67 @@
+3. `partitionby` : Specifies how the alerts will be partitioned.
+
+```bash
+fink_datatransfer \
+            -topic <topic name> \
+            -outdir <output directory> \
+            -partitionby time \
+            --verbose
+```
+In this new version, other options will need to be taken into account:
+1. `nconsumers` : Determine the number of processes (consumers) to use.\
+Each process will be assigned a partition in which it will consume all available alerts before being assigned a new, either partially processed or unprocessed partition.
+For example, with 10 partitions in our topic and 4 consumers, the first consumer will be assigned the first partition, the second consumer the second partition, and so on. Partition 5 will be assigned to the first one to finish retrieving all available alerts in its partition.
+In our examples, we used a computer with 8 cores. The program began experiencing crashes after using a number of consumers greater than 6. Therefore, it is recommended not to use a number of consumers higher than the available number of cores.
+2. `batchsize` : This is the number of alerts downloaded at once by a consumer, and it is set by default to 1000.
+
+```bash
+fink_datatransfer \
+            -topic <topic name> \
+            -outdir <output directory> \
+            -partitionby time \
+            -nconsumers 5 \
+            -batchsize 1000 \
+            --verbose
+```
+### Recommendations:
+The results obtained from using multiprocessing also depend on the computer's performance, including the number of CPU and Cores, memory cost, and available bandwidth of the network. Having a large number of consumers or a large batch size is not necessarily the best way to achieve good results.
+
+#### Nconsumers:
+Increasing the number of consumers can reduce the consumption time. However, the greater the number of consumers, the more demanding it is for the computer. Bandwidth should also be taken into account, as it will be shared among all processes, potentially decreasing speed for each additional process.\
+Here is an example in which we consume all the alerts from the month of July in Lightcurve (approximately 3.9 million alerts with a size of 5.23 GB). The alerts have been fully loaded into the topic before downloading.
+
+![1](/images/dt-1nconsumers.png)
+
+However, it is important to consider that the larger the number of consumers, the higher the chances of the program crashing, and the slower the individual consumer. 
+
+#### Batchsize:
+The batch size significantly influences the program's execution time, especially when the loading of alerts into the topic is complete. In this example with 5 `consumers`, we observe a significant decrease in execution time as the batch size increases.
+
+![1](/images/dt-1batchsize.png)
+
+It's worth noting that a larger batch size leads to a higher memory usage. Therefore, it should be moderated, as it could potentially cause crashes as well.
```

Comment:
I think we can be more concrete here. If I'm not mistaken:
1. Lightcurve option: assuming 1 alert is about 2KB, and a batchsize of 10000 -- this would correspond to ~20MB per batch per core 
2. Full packet option: assuming 1 alert is about 50KB, and a batchsize of 10000 -- this would correspond to ~500MB per batch per core (!)